templates/docs/patterns/hero/index.md

```diff
@@ -0,0 +1,82 @@
+---
+wrapper_template: '_layouts/docs.html'
+context:
+  title: Hero | Patterns
+---
+
+A hero is a prominent banner section typically used to quickly capture the user's attention after they land on the page.
+Depending on the size and composition of your content, you can choose from a variety of hero layouts.
+
```

Comment:
Would be good to include a section listing all the "parameters" of the pattern.

Not exactly in technical terms (as we don't document macro yet), but to let devs undestand what are the parts/options:

- layout - one of 50/50, etc
- title (mandatory) - displayed as h1 
- subtitle - displayed as h2-sized text
- description - free text
- image
- call for action block

It doesn't have to be exactly strict and acurate (in terms of what options are available in what layout, etc), just as an overview.

Reply:
It seems a bit too technical for now (we don't document the macro yet), so these options and flags don't really make sense to end user.

Right now all we need is a list of parameters and "slots":

Layout: defines the layout of the hero pattern, one of: … (list should ideally link to section on the page)
Title (required): Main title of the hero element, to put in `h1`
Subtitle: Optional subtitle, displayed as h2-sized text (because it's this is not h2 element, so let's not suggest that)
Description: …
Image: … 

I think it would be fine for now.

Reply:
Updated to be a less technical list of slots and params:

![image](https://github.com/user-attachments/assets/2e8a5ec9-f7f9-4167-b3f5-f8064c88a85c)

@lyubomir-popov the layouts list has also been made into a list of links to the sections for each layout as you [requested](https://github.com/canonical/vanilla-framework/pull/5242#issuecomment-2258290288)

Reply:
@pastelcyborg I think I understand the reason behind this (these content/layout choices are what makes the heros vary). Though I'm inclined to agree that documenting them as "slot" and "parameter" separately may be a bit confusing given that those are Jinja macro-specific terms and we are not yet documenting/exposing macros

Reply:
Yes, the intention for this was to already document the "spec" of the pattern: what are the building blocks of it, kind of the "anatomy" of the hero. I think it is valuable even without the macros, to understand what makes hero a hero, and in which ways the variants may differ.

Obviously, using the macro naming of parameters and slots makes it a bit confusing.

Maybe just make is a single list/table.

```
The hero patten is defined by:

- layout - one of … 
- title (mandatory) - h1 … 
- subtitle
- description
- call to action
- image
```

And drop the "is split on medium" from here for now, it doesn't really make sense yet on this level.

Reply:
@bartaz @pastelcyborg I've updated this inputs list accordingly:
![image](https://github.com/user-attachments/assets/7f1dd616-a49b-4bc3-9e59-85d99d0740a8)

Reply:
Adjusted the title, subtitle, description, cta, and image:

![image](https://github.com/user-attachments/assets/c7729e51-ffd6-458b-aa9a-8929b3527fb9)

Thinking about your other points now

Reply:
@pastelcyborg Updated to list the layouts outside the table, since the page overview nicely introduces layouts anyway. Let me know what you think

![image](https://github.com/user-attachments/assets/d936d226-974f-46d3-9a1c-67a62edf453c)

Reply:
Fine with me!